rules/numeric-host/src/lib.rs

```diff
@@ -0,0 +1,114 @@
+use anyhow::{anyhow, Ok};
+use bulwark_wasm_sdk::*;
+use lazy_static::lazy_static;
+use regex::Regex;
+
+struct NumericHostPlugin;
+
+lazy_static! {
+    static ref NUMERIC_HOST_REGEX: Regex =
+    Regex::new(r"^((?:(?:[0-9]+|0(x|X)[0-9a-fA-F]+|0(b|B)[0-7]+)\.){0,3}(?:[0-9]+|0(x|X)[0-9a-fA-F]+|0(b|B)[0-7]+)|\[[0-9a-fA-F:]+\])(:[0-9]{1,5})?$")
+        .unwrap();
+}
+
+fn check_numeric_host(req: Request) -> std::result::Result<bool, anyhow::Error> {
+    let host = req
+        .headers()
+        .get("Host")
+        .ok_or(anyhow!("Missing Host Header."))?;
+
+    let match_invalid_host = NUMERIC_HOST_REGEX.find(host.to_str()?);
+    Ok(match_invalid_host.is_some())
+}
```

Comment:
On reflection, I agree with the feedback about avoiding an error, though now I'm slightly worried about potential malicious use of invalid UTF-8 sequences as a bypass. Even in the current implementation, an error essentially represents a bypass, since the default behavior for unhandled errors is to no-op. Maybe the right behavior is to use [`from_utf8_lossy`](https://doc.rust-lang.org/std/string/struct.String.html#method.from_utf8_lossy) (maybe also filtering out "�" after conversion) and then parse, that way this function never returns an error. However, I'm inclined to address this as a separate issue post-merge rather than as part of this PR.